### ISSUE TITLE
Review long main-thread tasks

### ISSUE THREAD

**Issue by eons44** [web-infrastructure/doc_architecture#35]
Review long-running JavaScript tasks, which can cause the UI to "freeze".

Long-running tasks delay the time taken for a page to become interactive. This will impact your [Time to Interactive](https://web.dev/interactive/), a key measure of User Experience.
## How to use

This check lists the longest tasks on the main thread (i.e. blocking tasks), with the worst offenders listed first.

To improve your TTI, for any long-running tasks, consider:

* Can the script be removed or reduced in length?
* Can it be deferred until it is needed?
* Can it be run in parallel (i.e. off the main thread) via [Web Workers](https://css-tricks.com/off-the-main-thread/)?

## Further reading

[Are long JavaScript tasks delaying your Time to Interactive?](https://web.dev/long-tasks-devtools/) by Google
